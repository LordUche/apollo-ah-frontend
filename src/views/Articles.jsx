import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import time from '../utils/time';

/**
 * @description - returns all articles created by all authors
 * @param {props} articles - all authors article
 * @return {JSX}
 */
const Articles = ({ articles }) => (
  <div className="container">
    <div className="row">
      {articles.map(article => (
        <div key={article.id} className="col-lg-4 col-sm-6">
          <div className="article">
            <Link to="/single-article" className="link-surface" />
            <div className="article-cover">
              <img
                  src={article.image}
                  className="article-cover-img"
                  alt="article"
                />
            </div>
            <div className="article-item">
              <span className="article-category">{article.articleCategory.category}</span>
              <h4>{article.title}</h4>
              <p>
                {
                    article.description.length <= 50 ? article.description.substring(0, 50) : `${article.description.substring(0, 50)}...`
                  }
              </p>
              <div className="article-author">
                <img alt="article" src={article.User.Profile.image} />
                <div>
                  <h6>{article.User.Profile.username}</h6>
                  <span>
                    {`${time(article)} ago`}
                  </span>
                  <span className="article-author-dot" />
                  <span>{article.readTime}</span>
                  <i
                      className="fas fa-bookmark transition"
                    />
                </div>
              </div>
            </div>
          </div>
        </div>
      ))}
    </div>
  </div>
);

Articles.propTypes = {
  articles: PropTypes.array,
};

Articles.defaultProps = {
  articles: [],
};

export default Articles;
