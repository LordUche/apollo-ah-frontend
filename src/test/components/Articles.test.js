import React from 'react';
import { shallow } from 'enzyme';
import Articles from '../../views/Articles';

const articleProps = [{
  id: 1,
  image: 'https://images.unsplash.com/photo-1478358161113-b0e11994a36b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60',
  title: 'New article title',
  articleCategory: {
    category: 'Technology',
  },
  description: 'This is the article description',
  User: {id: 8, Profile:{firstname: 'Andra', lastname: 'Collins', username: '@Andra'}},
}];

describe('<Articles Test Suite>', () => {
  describe('<Articles>', () => {
    it('It should render succesfully', () => {
      const wrapper = shallow(<Articles articles={articleProps} />);
      expect(wrapper).toBeDefined();
      expect(wrapper.length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
