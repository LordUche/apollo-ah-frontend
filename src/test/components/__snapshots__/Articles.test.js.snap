// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Articles Test Suite> <Articles> It should render succesfully 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-4 col-sm-6"
      key="1"
    >
      <div
        className="article"
      >
        <Link
          className="link-surface"
          replace={false}
          to="/article/undefined"
        />
        <div
          className="article-cover"
        >
          <img
            alt="article"
            className="article-cover-img"
            src="https://images.unsplash.com/photo-1478358161113-b0e11994a36b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60"
          />
        </div>
        <div
          className="article-item"
        >
          <span
            className="article-category"
          >
            Technology
          </span>
          <h4>
            New article title
          </h4>
          <p>
            This is the article description
          </p>
          <div
            className="article-author"
          >
            <img
              alt="article"
            />
            <div>
              <h6>
                @Andra
              </h6>
              <span>
                almost NaN years ago
              </span>
              <span
                className="article-author-dot"
              />
              <span />
              <i
                className="fas fa-bookmark transition"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
